lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2506,8 +2489,10 @@ bool PrimalGen::performSynthesis(FunctionSynthesisItem item) {
   // Synthesize primal.
   auto substMap = item.original->getForwardingSubstitutionMap();
   if (auto primalGenEnv = item.target->getGenericEnvironment()) {
-    auto primalSubstMap = primalGenEnv->getForwardingSubstitutionMap();
-    substMap = substMap.subst(primalSubstMap);
+    substMap = SubstitutionMap::get(
+        primalGenEnv->getGenericSignature(),
+        QuerySubstitutionMap{primalGenEnv->getForwardingSubstitutionMap()},
```

Comment:
substMap is equivalent to just getForwardingSubstitutionMap() here

Reply:
I mean primalGenEnv->getForwardingSubstitutionMap()

Reply:
Thanks! Logic revamped in https://github.com/apple/swift/pull/22243/commits/0fb115dcc725e543b78d4994ace58eabf8ed69cf. (The old logic was simpler but led to stdlib compilation crash.)